src/app.js

```
@@ -1 +1,82 @@
+/* eslint-disable no-console */
 // write code here
+'use strict';
+
+const fs = require('fs/promises');
+const path = require('path');
+
+async function move(oldPath, newPath) {
+  try {
+    if (!oldPath || !newPath) {
+      throw new Error('Exactly two arguments are required');
+    }
+
+    if (oldPath === newPath) {
+      return;
+    }
+
+    const srcStats = await fs.stat(oldPath);
+
+    if (!srcStats.isFile()) {
+      throw new Error('Source is not a file');
+    }
+
+    const trailing = newPath.endsWith(path.sep) || newPath.endsWith('/');
+    let dest;
+
+    if (trailing) {
+      const destStats = await fs.stat(newPath);
+
+      if (!destStats.isDirectory()) {
+        throw new Error('Destination is not a directory');
+      }
+
+      dest = path.join(newPath, path.basename(oldPath));
+    } else {
```

Trailing-slash handling: you correctly detect trailing slash with newPath.endsWith(path.sep) || newPath.endsWith('/'). When `trailing` is true you stat newPath; ensure that if stat throws because the directory does not exist you allow that error to propagate (and cause a non-zero exit) rather than only logging it. With the top-level catch removed this will behave as required by the spec.